packages/environment-check/src/checks/environment.ts

```diff
@@ -137,26 +219,32 @@ async function getDestinationsResults(
  */
 export async function checkEnvironment(options?: CheckEnvironmentOptions): Promise<EnvironmentCheckResult> {
     const logger = getLogger();
+    let destinations: Destination[];
+    let destinationResults: { [dest: string]: DestinationResults };
 
     const { environment, messages } = await getEnvironment();
     logger.push(...messages);
 
-    const deepDiveDestinations = options?.destinations ? new Set(options.destinations) : new Set<string>();
-    if (options?.workspaceRoots?.length > 0) {
-        const workspaceResults = await getDestinationsFromWorkspace(options?.workspaceRoots);
-        logger.push(...workspaceResults.messages);
-        workspaceResults.destinations.forEach((dest) => deepDiveDestinations.add(dest));
-    }
-
-    const { messages: destMessages, destinations } = await checkBASDestinations();
-    logger.push(...destMessages);
+    if (isAppStudio()) {
+        const deepDiveDestinations = options?.destinations ? new Set(options.destinations) : new Set<string>();
+        if (options?.workspaceRoots?.length > 0) {
+            const workspaceResults = await getDestinationsFromWorkspace(options?.workspaceRoots);
+            logger.push(...workspaceResults.messages);
+            workspaceResults.destinations.forEach((dest) => deepDiveDestinations.add(dest));
+        }
 
-    const { messages: destDetailsMessages, destinationResults } = await getDestinationsResults(
-        deepDiveDestinations,
-        destinations,
-        options?.credentialCallback
-    );
-    logger.push(...destDetailsMessages);
+        const basDestResults = await checkBASDestinations();
```

Comment:
similar here, when we check destinations we need to add it to requested checks

```suggestion
        requestedChecks.add(Check.Destinations);
        const basDestResults = await checkBASDestinations();
```

Reply:
Updated